binding/default_validator.go

```diff
@@ -16,22 +18,57 @@ type defaultValidator struct {
 	validate *validator.Validate
 }
 
+type sliceValidateError []error
+
+func (err sliceValidateError) Error() string {
+	var errMsgs []string
+	for i, e := range err {
+		if e == nil {
+			continue
+		}
+		errMsgs = append(errMsgs, fmt.Sprintf("[%d]: %s", i, e.Error()))
+	}
+	return strings.Join(errMsgs, "\n")
+}
+
 var _ StructValidator = &defaultValidator{}
 
 // ValidateStruct receives any kind of type, but only performed struct or pointer to struct type.
 func (v *defaultValidator) ValidateStruct(obj interface{}) error {
+	if obj == nil {
+		return nil
+	}
+
 	value := reflect.ValueOf(obj)
 	valueType := value.Kind()
 	if valueType == reflect.Ptr {
-		valueType = value.Elem().Kind()
+		return v.ValidateStruct(value.Elem().Interface())
 	}
-	if valueType == reflect.Struct {
-		v.lazyinit()
-		if err := v.validate.Struct(obj); err != nil {
-			return err
+
+	switch valueType {
```

Comment:
use
```
switch value.Kind() {
    case reflect.Ptr:
...
```

Reply:
> use
> 
> ```
> switch value.Kind() {
>     case reflect.Ptr:
> ...
> ```

fixed in c29edbb8a059ba8c0271f6f45e3813dbf03b2aba